third_party/ibis/ibis_postgres/client.py

```diff
@@ -72,22 +72,23 @@ def connect(dbapi_connection, connection_record):
 
 
 def _metadata(self, query: str) -> sch.Schema:
```

Comment:
I think you need a "SELECT" check here similar to what happens in the Oracle client:

https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/7bfdfa14c87f9014cd700bc23a8b19312cd2a4e5/third_party/ibis/ibis_oracle/__init__.py#L139

Otherwise we can get an exception here when we drop in for a simple table, not a custom query:
```
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near ")"
LINE 1: select * from (dvt_test.tab_xml) t0 limit 0
```

I used the test table from issue https://github.com/GoogleCloudPlatform/professional-services-data-validator/issues/1330 to generate this example.

Reply:
Neil,
I am able to reproduce the issue - it looks like whenever you have a type that we don't recognize, then we throw an execption - even if that specific column is not part of the validation. I think this bug existed before the changes I made. We are off tomorrow - I will quickly check any response you provide.

Thanks.

Sundar Mudupalli

Reply:
Ok, I realize now under what conditions it will raise this error - and believe have made it more clear - it will result in NotImplementedError when types like xml are used that don't have equivalent Ibis types.

Sundar Mudupalli